### PR TITLE
Improve README to better explain both spawners

### PR DESCRIPTION
**Pull request by willingc**
The README is a bit light on details of DockerSpawner vs. SystemSpawner. Which to use? Why?

### CODE REVIEW

**Comment by willingc** [comment]
This PR:

- updates the README with more detail on DockerSpawner vs. SystemUserSpawner
- adds navigation
- provides more detail on installation and configuration steps
- adds a CONTRIBUTING.md file and instructions on creating a dev install.

File: README.md [hunk 1]
```diff
@@ -1,24 +1,84 @@
...
+In most cases, we recommend using DockerSpawner. Use cases where you
+may wish to use SystemUserSpawner are:
+
+- You are using docker just for environment management, but are running
+  on a system where the users already have accounts and files they
+  should be able to access from within the container. For example, you
+  wish to use the system users and user home directories that
+  already exist on a system.
+- You are using an external service that relies on distinct unix user
+  ownership and permissions (i.e. nbgrader).
```
Comment (akhmerov) [inline README.md]: i.e. -> e.g. / such as

**Comment by akhmerov** [comment]
Looks good. I also suggest to show how to specify the volume mapping for the DockerSpawner and explain that this is required for persistence. I believe such an extra remark is needed because username substitution is not an obvious part of machinery. Further, the warning about persistence should probably also be duplicated in the readme.

**Comment by willingc** [comment]
@akhmerov I added some text as suggested re: volume mapping and username substitution. Please take a look and let me know if I missed or misstated anything. Thanks!

**Comment by akhmerov** [comment]
Yes, looks great! I'm only confused about `/home/jovyan` vs `/home/jupyter`: both are used in different parts of readme.

**Comment by willingc** [comment]
@akhmerov Thanks for pointing that out. `/home/jovyan` and `/home/jupyter` is confusing. I've opted for `/home/jovyan` in the main README and went back and corrected in the DockerSpawner README.

**Comment by akhmerov** [comment]
@willingc that's all I see for now everything looks good. Good you saw that the username was also obsolete in the singleuser directory.

**Comment by willingc** [comment]
Thanks for all of your help and suggestions @akhmerov. :cake:

**Comment by minrk** [comment]
This is great! Thanks @willingc, and thanks @akhmerov for the review! :cake:

**Pull request merged**